plugins/prometheus/pkg/data/data.go

```diff
@@ -171,14 +184,18 @@ func GetMetrics(ctx context.Context, dynamicClient dynamic.Interface, restMapper
 		request.memoryLimit = memVal.Values[0].Value
 		combinedRequests[key] = request
 	}
-	requestArray := make([]CombinedRequest, 0, len(combinedRequests))
 
+	// Information about pods is required to lookup the number of containers
+	// each pod has, to manipulate per-pod network metrics to appear to be
+	// per-container.
 	client := controller.Client{
-		Context: ctx,
-		Dynamic: dynamicClient,
+		Context:    ctx,
+		Dynamic:    dynamicClient,
 		RESTMapper: restMapper,
 	}
-	workloads, err := client.GetAllTopControllersSummary("")
+	// This used to use client.GetAllTopControllersSummary(), which I believe
```

Comment:
@rbren I wanted to call this out and make sure you weigh in on this change since you updated other projects to use newer versions of our controller-utils. I believe we want to fetch Pods in the prometheus plugin, regardless of the new network metrics. THIs could also be a significant increase of resources pulled from in-cluster though.

Reply:
So both of these functions loop through all the pods in the cluster. The one you're using returns a _full_ list of pods, the old one just return (IIRC) a count and a single podspec.

It doesn't seem like you're using the individual pods, so maybe we can revert this?

Reply:
We had [a further discussion in Slack](https://fairwindsops.slack.com/archives/CNQBGRGFN/p1677668454068619) - some code here does iterate `workload.Pods`, we will keep the switch to `controller-utils.GetAllTopControllersWithPods`.